packages/parser/README.md

```diff
@@ -42,9 +42,17 @@ The following additional configuration options are available by specifying them
 
 - **`ecmaFeatures.jsx`** - default `false`. Enable parsing JSX when `true`. More details can be found [here](https://www.typescriptlang.org/docs/handbook/jsx.html).
 
-  - It's `false` on `*.ts` files regardless of this option.
-  - It's `true` on `*.tsx` files regardless of this option.
-  - Otherwise, it respects this option.
+  NOTE: this setting does not effect known file types (.js, .jsx, .ts, .tsx, .json) because the
+  typescript compiler has its own internal handling for known file extensions. The exact behaviour is as follows:
+
+  - if `parserOptions.project` is provided:
+    - `.js`, `.jsx`, `.tsx` files are parsed as if this is true.
+    - `.ts` files are parsed as if this is false.
+    - unknown extensions (`.md`, `.vue`) will respect this setting.
+  - if `parserOptions.project` is _not_ provided:
+    - `.js`, `.jsx`, `.tsx` files are parsed as if this is true.
+    - `.ts` files are parsed as if this is false.
+    - "unknown" extensions (`.md`, `.vue`) **are parsed as if this is false**.
```

Comment:
And `.vue` can be used with `jsx` syntax also.

Reply:
oops, I got the headings the wrong way around.

if you have a project, it will not respect the setting.
if you do not have a project, it will respect the setting.